lib/manticore/client.rb

```diff
@@ -610,23 +612,23 @@ def minimum_encoding_for(string)
 
     # Configure the SSL Context
     def ssl_socket_factory_from_options(ssl_options)
-      trust_store = trust_strategy = nil
+      trust_strategy = nil
 
       case ssl_options.fetch(:verify, :strict)
-      when false
-        trust_store = nil
-        trust_strategy = TrustSelfSignedStrategy::INSTANCE
-        verifier = SSLConnectionSocketFactory::ALLOW_ALL_HOSTNAME_VERIFIER
       when :disable, :none
-        trust_store = nil
         trust_strategy = TrustAllStrategy::INSTANCE
+        verifier = NoopHostnameVerifier::INSTANCE
+      when false # compatibility
+        trust_strategy = TrustSelfSignedStrategy::INSTANCE
         verifier = SSLConnectionSocketFactory::ALLOW_ALL_HOSTNAME_VERIFIER
```

Comment:
We are replacing another instance of deprecated `SSLConnectionSocketFactory::ALLOW_ALL_HOSTNAME_VERIFIER` in this PR with the equivalent `NoopHostnameVerifier::INSTANCE` -- should we get this one too?

```suggestion
        verifier = NoopHostnameVerifier::INSTANCE
```